### PR TITLE
DEV: De-jQ click-track

### PR DESCRIPTION
**Pull request by CvX**

### CODE REVIEW

File: app/assets/javascripts/discourse/app/lib/click-track.js [hunk 1]
```diff
@@ -11,27 +11,33 @@ import { selectedText } from "discourse/lib/utilities";
...
-  if ($link.is(".lightbox, .no-track-link, .hashtag, .back")) {
+  if (
+    ["lightbox", "no-track-link", "hashtag", "back"].some((name) =>
+      link.classList.contains(name)
+    )
+  ) {
     return false;
   }
 
   if ($link.parents("aside.quote, .elided, .expanded-embed").length !== 0) {
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/lib/click-track.js:26]: @jjaffeux what do we want to do with `$.parents()`? Add our own implementation to `dom-utils`?

Comment (davidtaylorhq) [reply app/assets/javascripts/discourse/app/lib/click-track.js:26]: How about

```suggestion
  if (link.closest("aside.quote, .elided, .expanded-embed")) {
```

Comment (CvX) [reply app/assets/javascripts/discourse/app/lib/click-track.js:26]: The difference between the two is:

`closest()` "Begins with the current element"
`parents()` "Begins with the parent element"

So for it to behave exactly the same, I guess I'd have to add an extra check:

```suggestion
  const closest = link.closest("aside.quote, .elided, .expanded-embed");
  if (closest && closest !== link) {
```

File: app/assets/javascripts/discourse/app/lib/click-track.js [hunk 2]
```diff
@@ -9,9 +9,18 @@ import { isTesting } from "discourse-common/config/environment";
...
-export function isValidLink($link) {
-  const link = $link[0];
+export function isValidLink(link) {
+  // eslint-disable-next-line no-undef
+  if (link instanceof jQuery) {
+    link = link[0];
+
+    deprecated("isValidLink now expects an Element, not a jQuery object", {
+      since: "2.9.0.beta7",
+      dropFrom: "2.10.0.beta1",
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/lib/click-track.js:21]: `3.0.0`

(internal `/t/3663/20`)

Comment (CvX) [reply app/assets/javascripts/discourse/app/lib/click-track.js:21]: I'm thinking about dropping `dropFrom`. It's not like we're actually going to remove it after the next release. 😛

**Review by davidtaylorhq — APPROVED** [review]

**Pull request merged**